File: update_namespaces.py
import os

base_dir = r"f:\MIllyass\pos-with-inventory-management\SourceCode\SQLAPI"

# 1. Change namespaces in POS.Data
data_dir = os.path.join(base_dir, "POS.Data")
for root, _, files in os.walk(data_dir):
    for file in files:
        if file.endswith(".cs"):
            filepath = os.path.join(root, file)
            with open(filepath, "r", encoding="utf-8") as f:
                content = f.read()
            
            new_content = content.replace("namespace POS.Domain", "namespace POS.Data")
            
            if new_content != content:
                with open(filepath, "w", encoding="utf-8") as f:
                    f.write(new_content)

# 2. Update usings globally
for root, _, files in os.walk(base_dir):
    for file in files:
        if file.endswith(".cs"):
            filepath = os.path.join(root, file)
            with open(filepath, "r", encoding="utf-8") as f:
                content = f.read()
            
            new_content = content.replace("using POS.Domain;", "using POS.Domain;\nusing POS.Data;")
            new_content = new_content.replace("using POS.Domain.Context;", "using POS.Data.Context;")
            new_content = new_content.replace("using POS.Domain.ImportExport;", "using POS.Data.ImportExport;")
            new_content = new_content.replace("using POS.Domain.FBR;", "using POS.Data.FBR;")
            new_content = new_content.replace("using POS.Domain.Sync;", "using POS.Data.Sync;")
            
            if new_content != content:
                with open(filepath, "w", encoding="utf-8") as f:
                    f.write(new_content)

print("Namespaces updated.")

File: update_namespaces.js
const fs = require('fs');
const path = require('path');

const baseDir = "f:\\MIllyass\\pos-with-inventory-management\\SourceCode\\SQLAPI";

function walkSync(dir, callback) {
    if (!fs.existsSync(dir)) return;
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        if (fs.statSync(filepath).isDirectory()) {
            walkSync(filepath, callback);
        } else {
            callback(filepath);
        }
    }
}

// 1. Change namespaces in POS.Data
const dataDir = path.join(baseDir, "POS.Data");
walkSync(dataDir, (filepath) => {
    if (filepath.endsWith(".cs")) {
        const content = fs.readFileSync(filepath, 'utf8');
        const newContent = content.replace(/namespace POS\.Domain/g, "namespace POS.Data");
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
        }
    }
});

// 2. Update usings globally
walkSync(baseDir, (filepath) => {
    if (filepath.endsWith(".cs")) {
        const content = fs.readFileSync(filepath, 'utf8');
        let newContent = content;
        
        // Don't add duplicate using POS.Data; if it's already there
        if (!newContent.includes("using POS.Data;")) {
            newContent = newContent.replace(/using POS\.Domain;/g, "using POS.Domain;\r\nusing POS.Data;");
        }
        
        newContent = newContent.replace(/using POS\.Domain\.Context;/g, "using POS.Data.Context;");
        newContent = newContent.replace(/using POS\.Domain\.ImportExport;/g, "using POS.Data.ImportExport;");
        newContent = newContent.replace(/using POS\.Domain\.FBR;/g, "using POS.Data.FBR;");
        newContent = newContent.replace(/using POS\.Domain\.Sync;/g, "using POS.Data.Sync;");
        
        // Also update DI references in Startup.cs or wherever they might be fully qualified
        newContent = newContent.replace(/POS\.Domain\.Context\.POSDbContext/g, "POS.Data.Context.POSDbContext");

        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
        }
    }
});

console.log("Namespaces updated.");
